esmf_utils/OutputInfo.F90

```diff
@@ -225,23 +250,26 @@ function make_ungridded_dim(info, n, rc) result(ungridded_dim)
       type(ESMF_Info), intent(in) :: info
       integer, optional, intent(out) :: rc
       integer :: status
-      character(len=:), allocatable :: key
       type(ESMF_Info) :: dim_info
+      character(len=ESMF_MAXSTR) :: raw
+      character(len=:), allocatable :: key
       character(len=:), allocatable :: name
       character(len=:), allocatable :: units
       real, allocatable :: coordinates(:)
       logical :: is_present
       character(len=1024) :: json_repr
 
       key = make_dim_key(n, _RC)
-      call ESMF_InfoGet(info, key=key, isPresent=is_present, _RC)
+      call ESMF_InfoGet(info, key=raw, isPresent=is_present, _RC)
       if(.not. is_present) then
          call ESMF_InfoPrint(info, unit=json_repr, _RC)
+         _FAIL('Key ' // trim(key) // ' not found in ' // trim(json_repr))
       end if
-      _ASSERT(is_present, 'Key ' // key // ' not found in ' // trim(json_repr))
-      dim_info = ESMF_InfoCreate(info, key=key, _RC)
-      call ESMF_InfoGetCharAlloc(dim_info, key=KEY_UNGRIDDED_NAME, value=name, _RC)
-      call ESMF_InfoGetCharAlloc(dim_info, key=KEY_UNGRIDDED_UNITS, value=units, _RC)
+      dim_info = ESMF_InfoCreate(info, key=trim(adjustl(raw)), _RC)
+      call ESMF_InfoGet(dim_info, key=KEY_UNGRIDDED_NAME, value=raw, _RC)
+      name = trim(adjustl(raw))
+      call ESMF_InfoGet(dim_info, key=KEY_UNGRIDDED_UNITS, value=raw, _RC)
+      units = trim(adjustl(raw))
```

Comment:
Have you attempted a reproducer to send to ESMF?

Reply:
I have not attempted a reproducer. With the `isPresent` check, `MAPL` throws an exception before the `ESMF_FieldGet` call. The original problem was that the `ESMF_Field` objects I used for the testing the History3G changes were missing the vertical dim spec. I wasn't using `isPresent`, so when I made the `ESMF_FieldGetCharAlloc` call I received an error about an allocatable character variable.

Reply:
I created a reproducer. The problem was what I described above. If the key is not found, the rc code is nonzero and the value becomes unallocated. Therefore it is not sufficient to allocate the value to an empty string (or any string) before calling `ESMF_InfoGetCharAlloc'.